exporters/prometheus/exporter.go

```diff
@@ -34,38 +37,47 @@ import (
 type Exporter struct {
 	metric.Reader
 	Collector prometheus.Collector
+
+	handler http.Handler
 }
 
+var _ metric.Reader = &Exporter{}
+
 // collector is used to implement prometheus.Collector.
 type collector struct {
-	metric.Reader
+	reader metric.Reader
 }
 
-// config is added here to allow for options expansion in the future.
-type config struct{}
+// New returns a Prometheus Exporter.
+func New(opts ...Option) (*Exporter, error) {
+	cfg := newConfig(opts...)
 
-// Option may be used in the future to apply options to a Prometheus Exporter config.
-type Option interface {
-	apply(config) config
-}
+	handler := promhttp.HandlerFor(cfg.gatherer, promhttp.HandlerOpts{})
```

Comment:
we should add WithHandlerOpts to our options as well if we are going to build this in.

Reply:
Do you have a specific use case?  If not maybe we add it when it's asked for.

Reply:
That is also fine.  People might want to use OpenMetrics, or might want to add the promhttp_metric_handler_errors_total metric for monitoring the handler itself.

Reply:
The workaround for this is to not use the exporter as the HttpHandler; use the Gatherer (or the DefaultGatherer) and create the HttpHandler yourself.

Reply:
What about replacing the `WithGatherer` option with something like the following?

```
func WithHandlerFor(gatherer prometheus.Gatherer, opts promhttp.HandlerOpts)
```

Reply:
If we go this route, I will also remove the HttpHandler methods.